torchtext/experimental/models/xlmr_model.py

```diff
@@ -0,0 +1,126 @@
+        >>> tensor([  903,    83,   142, 27781])
+    '''
+    tar_file = download_from_url(PRETRAINED['xlmr.regular'], root=root,
+                                 hash_value=MD5['xlmr.regular'], hash_type='md5')
+    checkpoint_file, tokenizer_file, vocab_file, args_file = extract_archive(tar_file, overwrite=True)
+    return _load_xlmr_model(checkpoint_file=checkpoint_file, args_file=args_file), _load_xlmr_transform(tokenizer_file=tokenizer_file, vocab_file=vocab_file)
+
+
+def _load_xlmr_model(checkpoint_file='model.pt', args_file='args.json'):
+    args = load_args_from_json(args_file)
+    encoder = TransformerEncoder(args.ntoken, args.embed_dim, args.nhead,
+                                 args.feedforward_dim, args.nlayers, args.dropout)
+    encoder.load_state_dict(torch.load(checkpoint_file))
+    return encoder
+
+
+def _load_xlmr_transform(tokenizer_file='sentencepiece.bpe.model', vocab_file='vocab.txt'):
+
+    tokenizer = sentencepiece_tokenizer(tokenizer_file)
+    with open(vocab_file, 'r') as f:
+        vocab = load_vocab_from_file(f)
+    return XLMRTransform(tokenizer, vocab)
+
+
+PRETRAINED = {'xlmr.regular': 'https://pytorch.s3.amazonaws.com/models/text/pretrained_models/xlmr.regular.tar.gz',
+              'xlmr.base': 'https://pytorch.s3.amazonaws.com/models/text/pretrained_models/xlmr.base.tar.gz'}
+MD5 = {'xlmr.regular': 'adf75f3d20c8a876533b206ccb3a7cb6',
+       'xlmr.base': 'abc0a28171f883c6e1b2e8cf184c1ce8'}
+
+##################################################################################
+# This part will be moved to stl-text/models folder
+
+
+class SentenceClassificationHead(nn.Module):
+    """Head for sentence-level classification."""
+
+    def __init__(self, num_labels, embed_dim=768, dropout=0.2):
+        super(SentenceClassificationHead, self).__init__()
+        self.dense = nn.Linear(embed_dim, embed_dim)
+        self.dropout = nn.Dropout(dropout)
+        self.out_proj = nn.Linear(embed_dim, num_labels)
+        self.activation = nn.Tanh()
+
+    def forward(self, input_features):
+        x = input_features[:, 0, :]  # The first token is reserved for [CLS]
+        x = self.dense(x)
+        x = self.activation(x)
+        x = self.dropout(x)
+        x = self.out_proj(x)
+        return x
+
+
+def _load_sentence_classifier(checkpoint_file='model.pt', args_file='args.json'):
+    args = load_args_from_json(args_file)
+    classifier = SentenceClassificationHead(args.num_labels, args.embed_dim, args.dropout)
+    classifier.load_state_dict(torch.load(checkpoint_file))
+    return classifier
+
+
+class TransformerEncoderSentenceClassification(nn.Module):
+    def __init__(self, transformer_encoder, classifier_head):
+        super(TransformerEncoderSentenceClassification, self).__init__()
+        self.transformer_encoder = transformer_encoder
+        self.classifier_head = classifier_head
+
+    def forward(src):
+        raise NotImplementedError("forward func has not been implemented yet.")
```

Comment:
I think it would be a good exercise to show the skeleton (like you have in here) of the abstractions that you are thinking about providing for the other tasks as well. It's not clear to me yet why this `TransformerEncoderSentenceClassification` implementation would be any different for a different task.